src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,99 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+        {
+            Debug.Assert(fieldHandle != null);
+            Debug.Assert(buffer != null);
+            Debug.Assert(bufferSize > 0);
+
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
+            {
+                bool supportedType = field.FieldType.Category switch
```

Comment:
What is this condition trying to do? I do not think it should be needed.

Reply:
@jkotas I probably misunderstood your previous request, this condition is replicating similar condition in JIT - `if (varTypeIsIntegral(lclTyp) || varTypeIsFloating(lclTyp) || (lclTyp == TYP_REF))` on NativeAOT side

Reply:
I am looking for the set of conditions in jitinterface.cpp and here to be roughly equivalent.

Reply:
Right, but there is no `field.GetFieldSize` on NativeAOT so I could filter out structs

Reply:
Ah, perhaps `field.FieldType.GetElementSize()` will do the work

Reply:
I do not think you need to have extra logic to filter structs. jitinterface.cpp is not filtering out structs either after the most recent feedback.

Reply:
> Ah, perhaps field.FieldType.GetElementSize() will do the work

The size of preinitialized data should be equal to the buffer size.